### PR TITLE
Feature/create target

### PR DESCRIPTION
**Pull request by t-romani**

### CODE REVIEW

File: app/controllers/api/v1/targets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module Api
+  module V1
+    class TargetsController < Api::V1::ApiController
+      helper_method :user
+
+      def create
+        @target = Target.create! target_params
```
Comment (gonzaM) [inline app/controllers/api/v1/targets_controller.rb]: Use parenthesis

File: app/controllers/api/v1/targets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        @target.save
```
Comment (gonzaM) [inline app/controllers/api/v1/targets_controller.rb]: You don't need to save, `create!` already saves the record

File: app/models/target.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#
+#  fk_rails_...  (topic_id => topics.id)
+#  fk_rails_...  (user_id => users.id)
+#
+
+class Target < ApplicationRecord
+  belongs_to :topic
+  belongs_to :user
+
+  validates :latitude, :longitude, :radius, :topic_id, :user_id, presence: true
```
Comment (gonzaM) [inline app/models/target.rb]: It's not necessary to validate the presence of belgons_to associations, it's made by default in rails >= 5

File: app/views/api/v1/targets/_info.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,13 @@
+json.id           target.id
+json.topic_id     target.topic_id
+json.title        target.title
+json.radius       target.radius
+json.latitude     target.latitude
+json.longitude    target.longitude
+json.created_at   target.created_at
+json.updated_at   target.updated_at
+json.user do
```
Comment (gonzaM) [inline app/views/api/v1/targets/_info.json.jbuilder]: Don't you have a user partial?
Also, if you will use `extract!`, use it here too

File: spec/models/target_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+require 'rails_helper'
+
+describe Target, type: :model do
+  let!(:user)          { create(:user) }
+
+  subject { build(:target, user: user) }
+
+  describe 'presence validations' do
+    it { should validate_presence_of(:title) }
```
Comment (gonzaM) [inline spec/models/target_spec.rb]: Use new rspec syntax here

File: spec/models/target_spec.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    it { should validate_presence_of(:latitude) }
+    it { should validate_presence_of(:longitude) }
+    it { should validate_presence_of(:radius) }
+    it { should validate_presence_of(:topic_id) }
+    it { should validate_presence_of(:user_id) }
+  end
+
+  describe 'belonging validations' do
+    it { should belong_to(:user) }
```
Comment (gonzaM) [inline spec/models/target_spec.rb]: Same here. And also, I would refactor the describes

File: spec/request/target/create_spec.rb [hunk 7]
```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+describe 'POST #create target', type: :request do
+  let!(:user)           { create(:user) }
+  let!(:auth_headers)   { auth_user_headers }
+  let!(:topic)          { create(:topic) }
+  let!(:target_params)  {
```
Comment (gonzaM) [inline spec/request/target/create_spec.rb]: Use `do...end` for multiline blocks

File: app/models/target.rb (hunk 3 above)
Comment (t-romani) [reply app/models/target.rb]: @gonzaM  does it?
my presence validation tests fail if I delete the presence validation on the model

File: spec/models/target_spec.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+# Foreign Keys
+#
+#  fk_rails_...  (topic_id => topics.id)
+#  fk_rails_...  (user_id => users.id)
+#
+
+require 'rails_helper'
+
+describe Target, type: :model do
+  let!(:user)          { create(:user) }
```
Comment (gonzaM) [inline spec/models/target_spec.rb]: unnecessary spaces

File: spec/request/target/create_spec.rb [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      end
+
+      it 'does not create the target' do
+        expect { subject }.not_to change(Target, :count)
+      end
+    end
+
+    context 'when missing argument' do
+      context 'title' do
+        let(:data) { parsed_data}
```
Comment (gonzaM) [inline spec/request/target/create_spec.rb]: missing space

Comment (gonzaM) [reply spec/request/target/create_spec.rb]: Is it necessary to define this? Couldn't you use `parsed_data` directly?

**Review by gonzaM — APPROVED** [review]

File: db/migrate/20190827162435_create_targets.rb [hunk 10]
```diff
@@ -0,0 +1,16 @@
+class CreateTargets < ActiveRecord::Migration[5.2]
+  def change
+    create_table :targets do |t|
+      t.belongs_to  :topic, null: false
+      t.belongs_to  :user, null: false
+      t.string      :title, null: false
+      t.float       :radius, :latitude, :longitude, null: false
```
Comment (JuanfraM) [inline db/migrate/20190827162435_create_targets.rb]: I prefer to keep columns separate

File: db/migrate/20190827162435_create_targets.rb [hunk 11]
```diff
@@ -0,0 +1,16 @@
+class CreateTargets < ActiveRecord::Migration[5.2]
+  def change
+    create_table :targets do |t|
+      t.belongs_to  :topic, null: false
```
Comment (JuanfraM) [inline db/migrate/20190827162435_create_targets.rb]: I'd use `t.references` for both cases. Also this creates the foreign key
[check this](https://stackoverflow.com/questions/16309742/t-references-in-the-migration-vs-belongs-to-in-the-model/16309795)

File: spec/request/target/create_spec.rb [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+        it 'gets an bad request response' do
+          subject
+          expect(response).to have_http_status(:bad_request)
+        end
+
+        it 'returns error message' do
+          subject
+          expect(parsed_data['error']).to eq(
+            'Validation failed: Title can\'t be blank'
```
Comment (JuanfraM) [inline spec/request/target/create_spec.rb]: `"Validation failed: Title can't be blank"`

**Review by JuanfraM — APPROVED** [review]
Some minor comments

**Pull request merged**